components/src/core/IconSwitch/IconSwitch.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import { StyleSheet, TouchableOpacity, ViewProps, View } from 'react-native';
+import Animated, {
+    interpolate,
+    useAnimatedStyle,
+    useSharedValue,
+    withTiming,
+    Easing,
+    Extrapolation,
+    interpolateColor,
+} from 'react-native-reanimated';
+import { Icon } from '../Icon/Icon';
+import { $DeepPartial } from '@callstack/react-theme-provider';
+import { MD3Theme, useTheme } from 'react-native-paper';
+
+export type IconSwitchProps = ViewProps & {
+    checkedIcon?: boolean;
+    disabledSwitch?: boolean;
+    color?: string;
+    value?: boolean;
+    onValueChange?: Function;
+    theme?: $DeepPartial<MD3Theme>;
+};
```

Comment:
You will want to ensure that your prop names reflect how they are meant to be used. For example, I cannot tell by just looking at the name whether `checkedIcon` is asking me to pass an IconSource to specify what icon to use, or it is simply a boolean property to toggle icons on or off. `disabledSwitch` is also inconsistent with how this prop is named in other components.

Reply:
I have changed `changedIcon` name to `showIcon`, it is a boolean property to toggle icons on or off. Is it fine?

Reply:
That's better, thank you